src/tracks/playabletrack/wavetrack/ui/ClipOverflowButtonHandle.cpp

```diff
@@ -0,0 +1,103 @@
+#include "ClipOverflowButtonHandle.h"
+#include "BasicUI.h"
+#include "ClipParameters.h"
+#include "HitTestResult.h"
+#include "ProjectAudioIO.h"
+#include "ProjectHistory.h"
+#include "ProjectWindows.h"
+#include "RefreshCode.h"
+#include "UndoManager.h"
+#include "WaveClip.h"
+#include "WaveClipUtilities.h"
+#include "WaveTrack.h"
+#include "ZoomInfo.h"
+
+namespace
+{
+constexpr auto buttonWidth = 24;
+
+wxRect GetButtonRectangle(const wxRect& affordanceRect)
+{
+   const auto x = affordanceRect.x + affordanceRect.width - buttonWidth;
+   return { x, affordanceRect.y, buttonWidth, affordanceRect.height };
+}
+} // namespace
+
+ClipOverflowButtonHandle::ClipOverflowButtonHandle(
+   const std::shared_ptr<WaveTrack>& track,
+   const std::shared_ptr<WaveTrack::Interval>& clip)
+    : mTrack { track }
+    , mClip { clip }
+{
+}
+
+bool ClipOverflowButtonHandle::HitTest(
+   const ClipTimes& clip, const ZoomInfo& viewInfo, const wxRect& viewRect,
+   const wxPoint& point)
+{
+   const auto clipRect = ClipParameters::GetClipRect(clip, viewInfo, viewRect);
+   return GetButtonRectangle(clipRect).Contains(point);
+}
+
+wxRect
+ClipOverflowButtonHandle::DrawOnClip(wxDC& dc, const wxRect& affordanceRect)
+{
+   const auto rect = GetButtonRectangle(affordanceRect);
+   // Invisible pen
+   constexpr auto penWidth = 0;
+   wxDCPenChanger pen { dc, { wxPenInfo { wxColour {}, penWidth } } };
```

Comment:
For me, the debug build was violating an assertion in wxWidgets, but my build didn't have debug symbols and I couldn't see wxWidgets source code.

After some trial and error, I figured out that replacing `wxColour{}` with `*wxBLACK` was a sufficient fix.